include/swift/AST/ASTScope.h

```diff
@@ -88,6 +91,12 @@ struct AnnotatedInsertionPoint {
   ASTScopeImpl *insertionPoint;
   const char *explanation;
 };
+} // namespace ast_scope
+
+SourceLoc extractNearestSourceLoc(
+    std::tuple<ast_scope::ASTScopeImpl *, ast_scope::ScopeCreator *>);
```

Comment:
Declare this in namespace `ast_scope`, so you (1) won't need to qualify `ASTScopeImpl` and `ScopeCreator`, and (2) won't have the include-ordering dependency, because `extractNearestSourceLoc` will be found by Argument Dependent Lookup (ADL).